addons/loyalty/models/loyalty_program.py

```diff
@@ -577,3 +577,15 @@ def _get_template_values(self):
                 })]
             },
         }
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        """
+        trigger_product_ids will overwrite product ids defined in a loyalty rule in certain instances. Thus, it should
+        be explicitly removed from an incoming vals dict unless, of course, it was actually a visible field.
+        """
+        for vals in vals_list:
+            if 'trigger_product_ids' in vals and vals['program_type'] not in ['gift_card', 'ewallet']:
```

Comment:
@ethanrobv 
Since there is a default value, `program_type` may not be in vals either, leading to a traceback. Need something like
```suggestion
            if 'trigger_product_ids' in vals and 'program_type' in vals and vals['program_type'] not in ['gift_card', 'ewallet']:
```